CollectNewspaperKarel.java

```diff
@@ -1,17 +1,57 @@
-/*
- * File: CollectNewspaperKarel.java
- * --------------------------------
- * At present, the CollectNewspaperKarel subclass does nothing.
- * Your job in the assignment is to add the necessary code to
- * instruct Karel to walk to the door of its house, pick up the
- * newspaper (represented by a beeper, of course), and then return
- * to its initial position in the upper left corner of the house.
- */
-
-import stanford.karel.*;
-
-public class CollectNewspaperKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CollectNewspaperKarel.java
+ * --------------------------------
+ * At present, the CollectNewspaperKarel subclass does nothing.
+ * Your job in the assignment is to add the necessary code to
+ * instruct Karel to walk to the door of its house, pick up the
+ * newspaper (represented by a beeper, of course), and then return
+ * to its initial position in the upper left corner of the house.
+ */
+//
+//
+import stanford.karel.*;
+
+public class CollectNewspaperKarel extends SuperKarel {
+
+//The Program is to make Karel pick up the newspaper and return to its primary position
+	
+	public void run() {
+		moveToTheNewspaper();

```

Comment:
Great job on breaking down the program in smaller parts that can be understood by any other developer.